source/speech.py

```diff
@@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
 			res.append(symbol)
 		else:
 			res.append(u"{count} {symbol}".format(count=count, symbol=symbol))
-
-	return " ".join(res)
+		quarterTones += (count*4 if raw[0]== "\t" else count)

```

Comment:
This code uses raw[0] and len(raw) which is suggesting raw is a list.

raw = m.group()
         symbol = characterProcessing.processSpeechSymbol(locale, raw[0])
         count = len(raw)
         if symbol == raw[0]:


On 6/9/2016 4:44 AM, James Teh wrote:
>
> In source/speech.py 
> <https://github.com/nvaccess/nvda/pull/6057#discussion_r66418995>:
>
> > @@ -436,8 +441,16 @@ def getIndentationSpeech(indentation):
> >  			res.append(symbol)
> >  		else:
> >  			res.append(u"{count} {symbol}".format(count=count, symbol=symbol))
> > -
> > -	return " ".join(res)
> > +		quarterTones += (count*4 if raw[0]== "\t" else count)
>
> Out of interest, why the |[0]|? At this point, we should already be 
> dealing with single characters... and even if we weren't in some 
> future/alternative universe, that suggests the space is a double byte 
> character or something, so the second character could be important.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/pull/6057/files/149618ac951d0380dad97bf4bd02ae1f00e451f9#r66418995>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivZoxeDuVjabSpTjZYxA_GKouq6R5ks5qJ-6igaJpZM4IxqMD>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194